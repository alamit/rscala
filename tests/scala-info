#!/bin/bash

# V210=2.10.5
V211=2.11.8

killall java > /dev/null 2>&1
R --slave -e 'library(rscala); print(scalaInfo("~/local/scala/current",verbose=TRUE)$cmd)'

killall java > /dev/null 2>&1
R --slave -e 'library(rscala); print(scalaInfo("~/local/scala/scala-'$V211'",verbose=TRUE)$cmd)'

# killall java > /dev/null 2>&1
# R --slave -e 'library(rscala); print(scalaInfo("~/local/scala/scala-'$V210'",verbose=TRUE)$cmd)'


killall java > /dev/null 2>&1
SCALA_HOME=~/local/scala/current R --slave -e 'library(rscala); print(scalaInfo(verbose=TRUE)$cmd)'

killall java > /dev/null 2>&1
SCALA_HOME=~/local/scala/scala-$V211 R --slave -e 'library(rscala); print(scalaInfo(verbose=TRUE)$cmd)'

# killall java > /dev/null 2>&1
# SCALA_HOME=~/local/scala/scala-$V210 R --slave -e 'library(rscala); print(scalaInfo(verbose=TRUE)$cmd)'


killall java > /dev/null 2>&1
SCALA_HOME= PATH=~/local/scala/current/bin:$PATH R --slave -e 'library(rscala); print(scalaInfo(verbose=TRUE)$cmd)'

killall java > /dev/null 2>&1
SCALA_HOME= PATH=~/local/scala/scala-$V211/bin:$PATH R --slave -e 'library(rscala); print(scalaInfo(verbose=TRUE)$cmd)'

# killall java > /dev/null 2>&1
# SCALA_HOME= PATH=~/local/scala/scala-$V210/bin:$PATH R --slave -e 'library(rscala); print(scalaInfo(verbose=TRUE)$cmd)'

mv ~/.rscala ~/.rscala-off
killall java > /dev/null 2>&1
SCALA_HOME= PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games ~/local/R/current/bin/R --slave -e 'library(rscala)'

killall java > /dev/null 2>&1
mv ~/.rscala-off ~/.rscala
SCALA_HOME= PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games ~/local/R/current/bin/R --slave -e 'library(rscala); scalaInstall()'
SCALA_HOME= PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games ~/local/R/current/bin/R --slave -e 'library(rscala); print(scalaInfo(verbose=TRUE)$cmd)'

