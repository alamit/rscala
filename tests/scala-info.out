
Searching for a suitable Scala installation.
  * ATTEMPT: Found a candidate (/home/dahl/local/scala/scala-2.11.8/bin/scala)
  * SUCCESS: 'scala.home' (~/local/scala/current) argument

[1] "/home/dahl/local/scala/scala-2.11.8/bin/scala"

Searching for a suitable Scala installation.
  * ATTEMPT: Found a candidate (/home/dahl/local/scala/scala-2.11.8/bin/scala)
  * SUCCESS: 'scala.home' (~/local/scala/scala-2.11.8) argument

[1] "/home/dahl/local/scala/scala-2.11.8/bin/scala"

Searching for a suitable Scala installation.
  * FAILURE: 'scala.home' argument is NULL
  * ATTEMPT: Found a candidate (/home/dahl/local/scala/scala-2.11.8/bin/scala)
  * SUCCESS: SCALA_HOME (/home/dahl/local/scala/current) environment variable

[1] "/home/dahl/local/scala/scala-2.11.8/bin/scala"

Searching for a suitable Scala installation.
  * FAILURE: 'scala.home' argument is NULL
  * ATTEMPT: Found a candidate (/home/dahl/local/scala/scala-2.11.8/bin/scala)
  * SUCCESS: SCALA_HOME (/home/dahl/local/scala/scala-2.11.8) environment variable

[1] "/home/dahl/local/scala/scala-2.11.8/bin/scala"

Searching for a suitable Scala installation.
  * FAILURE: 'scala.home' argument is NULL
  * FAILURE: SCALA_HOME () environment variable
  * ATTEMPT: Found a candidate (/home/dahl/local/scala/scala-2.11.8/bin/scala)
  * SUCCESS: 'scala' in the shell's search path

[1] "/home/dahl/local/scala/scala-2.11.8/bin/scala"

Searching for a suitable Scala installation.
  * FAILURE: 'scala.home' argument is NULL
  * FAILURE: SCALA_HOME () environment variable
  * ATTEMPT: Found a candidate (/home/dahl/local/scala/scala-2.11.8/bin/scala)
  * SUCCESS: 'scala' in the shell's search path

[1] "/home/dahl/local/scala/scala-2.11.8/bin/scala"

Searching for a suitable Scala installation.
  * FAILURE: 'scala.home' argument is NULL
  * FAILURE: SCALA_HOME () environment variable
  * FAILURE: 'scala' in the shell's search path
  * FAILURE: special installation directory (/home/dahl/.rscala/scala-2.11.8)
Cannot find a suitable Scala installation.

Scala (>= 2.10) is not distributed with the rscala package.  Individuals can
install Scala using "rscala::scalaInstall()" which places Scala in "~/.rscala"
in the user's home directory.  System administrators should install Scala
globally as described here: http://www.scala-lang.org/download/install.html.
In short, simply download the archive, unpack it, and add the "scala" script to
the path.  Note that Scala itself requires Java.  The rscala package tries
various techniques to find a suitable Scala installation, including searching
the path.  Details on the search is available using
"rscala::scalaInfo(verbose=TRUE)".

trying URL 'http://downloads.typesafe.com/scala/2.11.8/scala-2.11.8.tgz'
Content type 'application/octet-stream' length 28678231 bytes (27.3 MB)
==================================================
downloaded 27.3 MB

Successfully installed Scala in /home/dahl/.rscala/scala-2.11.8

Searching for a suitable Scala installation.
  * FAILURE: 'scala.home' argument is NULL
  * FAILURE: SCALA_HOME () environment variable
  * FAILURE: 'scala' in the shell's search path
  * ATTEMPT: Found a candidate (/home/dahl/.rscala/scala-2.11.8/bin/scala)
  * SUCCESS: special installation directory (/home/dahl/.rscala/scala-2.11.8)

[1] "/home/dahl/.rscala/scala-2.11.8/bin/scala"
